src/NzbDrone.Core/Download/Clients/qBittorrent/QBittorrentProxy.cs

```diff
@@ -94,12 +94,26 @@ public void RemoveTorrent(string hash, Boolean removeData, QBittorrentSettings s
 
         public void SetTorrentLabel(string hash, string label, QBittorrentSettings settings)
         {
-            var request = BuildRequest(settings).Resource("/command/setLabel")
-                                                .Post()
-                                                .AddFormParameter("hashes", hash)
-                                                .AddFormParameter("label", label);
-
-            ProcessRequest<object>(request, settings);
+            var setCategoryRequest = BuildRequest(settings).Resource("/command/setCategory")
+                                                        .Post()
+                                                        .AddFormParameter("hashes", hash)
+                                                        .AddFormParameter("category", label);
+            try
+            { 
+                ProcessRequest<object>(setCategoryRequest, settings);
+            }
+            catch(DownloadClientException ex)
+            {
+                // if setCategory fails due to method not being found, then try older setLabel command for qbittorent < v.3.3.5
+                if (ex.InnerException is HttpException && (ex.InnerException as HttpException).Response.StatusCode == HttpStatusCode.NotFound)
```

Comment:
Since `as` won't throw if its not that type, we would use the following:

```
var httpException = ex.InnerException as HttpException;

if (httpException && httpException.Response.StatusCode == HttpStatusCode.NotFound)
```

It makes it a little more clear as its spread out a bit. With C# 6 I think we could even do:

```
if ((ex.InnerException as HttpException)?.Response.StatusCode == HttpStatusCode.NotFound)
```

(https://msdn.microsoft.com/en-CA/library/dn986595.aspx)